addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5516,9 +5516,8 @@ registry.ReplaceMedia = SnippetOptionWidget.extend({
      * @see this.selectClass for parameters
      */
     async replaceMedia() {
-        // TODO for now, this simulates a double click on the media,
-        // to be refactored when the new editor is merged
-        this.$target.dblclick();
+        // open mediaDialog and replace the media.
+        await this.options.wysiwyg.openMediaDialog(this.$target[0]);
```

Comment:
See `openMediaDialog`'s function signature.
The correct way to pass a node to it would be 
```js
await this.options.wysiwyg.openMediaDialog({ node:this.$target[0] });
```